tests/unit/test_hnswlib.cpp

```diff
@@ -98,44 +100,60 @@ TEST_F(HNSWLibTest, resizeNAlignIndex_initailCapacity_is_big) {
 
     float a[dim];
 
-    // add up to blocksize + 1
+    // add up to blocksize + 1 = 3 + 1 = 4
     for (size_t i = 0; i < bs + 1; i++) {
         for (size_t j = 0; j < dim; j++) {
             a[j] = (float)i;
         }
         VecSimIndex_AddVector(index, (const void *)a, i);
     }
 
-    // the capcity shouldnt be changed
+    // The capcity shouldnt be changed. (32)
     ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity(), n);
 
-    // delete random vector, to get size % block_size == 0
+    // Delete random vector, to get size % block_size == 0. size = 3
     VecSimIndex_DeleteVector(index, 1);
 
-    ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexSize(), bs);
+    // Index size = bs = 3.
+    ASSERT_EQ(VecSimIndex_IndexSize(index), bs);
 
-    // the capacity should be now aligned with the block size
+    // The capacity should be now aligned with the block size.
     ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity() % bs, 0);
-    // new capacity = intiailcapacity - block_size - number_of_vectors_to_align
+    // New capacity = intiailcapacity - block_size - number_of_vectors_to_align =
+    // 32  - 3 - 32 % 3 (2) = 27
     ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity(),
               n - bs - n % bs);
+    // Check actual numbers **cahnge this if you change the intial parameters**
+    ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity(), 27);
 
     size_t curr_capacity = reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity();
```

Comment:
To have cleaner code, you can move this to be before line 124, and test instead:
`ASSERT_EQ(curr_capacity, n - bs - n % bs);`